### PR TITLE
Update docs remove dry_run in docstring of file.blockreplace state.

### LINKED ISSUE #60227: [DOCS] `dry_run` in docstring of `file.blockreplace` state, but not implemented
**Description**
`dry_run` kwarg is documented in docstring of `file.blockreplace` state, but not implemented

**Suggested Fix**
Remove `dry_run` from the docstring

**Type of documentation**
Salt Repo

**Location or format of documentation**
https://github.com/saltstack/salt/blob/v3003/salt/states/file.py#L5614
https://docs.saltproject.io/en/3003/ref/states/all/salt.states.file.html#salt.states.file.blockreplace

**Additional context**
`blockreplace` seems to be the only function in the file module that still has this kwarg documented.

### PR DESCRIPTION
**Pull request by ericham**
### What does this PR do?
Update docs remove dry_run in docstring of file.blockreplace state.

### What issues does this PR fix or reference?
Fixes: #60227 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by ericham** [comment]
@barbaricyawps here is my attempt at this doc update.

File: doc/man/salt.7 [hunk 1]
```diff
@@ -403017,10 +403017,6 @@ New in version 3001.
 The file extension to use for a backup of the file if any edit is made.
 Set this to \fBFalse\fP to skip making a backup.
 .TP
-.B dry_run
-If \fBTrue\fP, do not make any edits to the file and simply return the
-changes that \fIwould\fP be made.
-.TP
```
Comment (MKLeb) [inline doc/man/salt.7:403023]: We don't need to update the man pages, that is done during the release process.  Would you mind reverting these changes?

Comment (ericham) [reply doc/man/salt.7]: Thanks for the feedback, it's my first time doing a doc update and I just did a git grep to catch possible changes needed. I've reverted the man page change.

**Review by MKLeb — APPROVED** [review]

**Comment by MKLeb** [comment]
@ericham Nope, sorry.  I'll see when we can get this merged.  Thanks for the contribution!

**Pull request merged**